src/utils/types/FormValidationStatus.ts

```diff
@@ -1 +1 @@
-export type FormValidationStatus = 'error' | 'success'
+export type FormValidationStatus = 'error' | 'success' | 'warning' // TODO: remove warning as a breaking change. Added for backwards compatibility with TextArea
```

Comment:
Not ideal, but this avoids a breaking change to TextInput by leaving `warning` in play. I'd recommend we use the upcoming form validation work to remove `warning` altogether and defer the breaking change.